profitbricks/src/main/java/org/jclouds/profitbricks/domain/LoadBalancer.java

```diff
@@ -0,0 +1,398 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import org.jclouds.javax.annotation.Nullable;
+
+import java.util.Date;
+import java.util.List;
+
+@AutoValue
+public abstract class LoadBalancer {
+
+   public enum LoadbalancerAlgorithm {
+
+      ROUND_ROBIN, UNRECOGNIZED;
+
+      public static LoadbalancerAlgorithm
+              fromValue(String value) {
+         try {
+            return valueOf(value);
+         } catch (IllegalArgumentException e) {
+            return UNRECOGNIZED;
+         }
+      }
+   }
+
+   @Nullable
+   public abstract String id();
+
+   @Nullable
+   public abstract String name();
+
+   @Nullable
+   public abstract LoadbalancerAlgorithm loadBalancerAlgorithm();
+
+   @Nullable
+   public abstract String dataCenterId();
+
+   @Nullable
+   public abstract String dataCenterVersion();
+
+   @Nullable
+   public abstract boolean internetAccess();
+
+   @Nullable
+   public abstract String ip();
+
+   @Nullable
+   public abstract String lanId();
+
+   @Nullable
+   public abstract ProvisioningState state();
+
+   @Nullable
+   public abstract Date creationTime();
+
+   @Nullable
+   public abstract Date lastModificationTime();
+
+   @Nullable
+   public abstract List<Server> balancedServers();
+
+   @Nullable
+   public abstract List<Firewall> firewalls();
+
+   public static LoadBalancer create(String id, String name, LoadbalancerAlgorithm loadBalancerAlgorithm,
+           String dataCenterId, String dataCenterVersion, boolean internetAccess,
+           String ip, String lanId, ProvisioningState state, Date creationTime, Date lastModificationTime, List<Server> balancedServers, List<Firewall> firewalls) {
+      return new AutoValue_LoadBalancer(id, name, loadBalancerAlgorithm, dataCenterId, dataCenterVersion, internetAccess, ip, lanId, state, creationTime, lastModificationTime,
+              balancedServers != null ? ImmutableList.copyOf(balancedServers) : ImmutableList.<Server>of(),
+              firewalls != null ? ImmutableList.copyOf(firewalls) : ImmutableList.<Firewall>of());
+   }
+
+   public static Builder builder() {
+      return new Builder();
+   }
+
+   public static class Builder {
+
+      public String id;
+
+      public String name;
+
+      public LoadbalancerAlgorithm loadBalancerAlgorithm;
+
+      public String dataCenterId;
+
+      public String dataCenterVersion;
+
+      public boolean internetAccess;
```

Comment:
Change this to `Boolean`. Using `boolean` will make it default to false, subsequently adding it in the request binder. (Might not be the behavior we're expecting). Change the builder method below as well